pickle.c

```diff
@@ -885,16 +905,12 @@ static int picolUnEscape(char *inout) {
 			case  ']': r[k] = ']';  break;
 			case  'e': r[k] = 27;   break;
 			case  'x': {
-				if (!inout[j + 1] || !inout[j + 2])
-					return -1;
-				const char v[3] = { inout[j + 1], inout[j + 2], 0 };
-				unsigned d = 0;
-				int pos = 0;
-				if (sscanf(v, "%x%n", &d, &pos) != 1)
+				int val;
+				const int pos = hexstr2toint(&inout[j + 1], &val);
```

Comment:
Again I would assign an initial value to val, in case 'hexstr2toint' does not assign one, or is modified so it does not.